src/k8s/pkg/k8sd/controllers/csrsigning/controller.go

```diff
@@ -19,14 +19,14 @@ import (
 
 type Controller struct {
 	snap      snap.Snap
-	waitReady func()
+	waitReady func(context.Context) error
```

Comment:
rather than this, since we have the `waitKubernetesReady()` method, let's block on that and then only start the controllers after it's done. therefore, we do not need to pass the `waitReady` to each individual controller.